modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/WorkspaceExtension.java

```diff
@@ -272,6 +321,111 @@ public void setTargetPlatformVersion(Object targetPlatformVersion) {
 		_targetPlatformVersion = targetPlatformVersion;
 	}
 
+	private String _getDefaultAppServerVersion() {
+		if (_productInfo != null) {
+			return _productInfo.getAppServerTomcatVersion();
+		}
+
+		return null;
+	}
+
+	private String _getDefaultDockerImage() {
+		String defaultDockerImage = _DOCKER_IMAGE_LIFERAY;
+
+		if (_productInfo != null) {
+			defaultDockerImage = _productInfo.getLiferayDockerImage();
+		}
+
+		return defaultDockerImage;
+	}
+
+	private String _getDefaultLiferayProductVersion() {
+		if (_productInfo != null) {
+			return _productInfo.getLiferayProductVersion();
+		}
+
+		return null;
+	}
+
+	private String _getDefaultProductBundleUrl() {
+		String defaultBundleUrl = BundleSupportConstants.DEFAULT_BUNDLE_URL;
+
+		if (_productInfo != null) {
+			Base64.Decoder decoder = Base64.getDecoder();
+
+			defaultBundleUrl = new String(
+				decoder.decode(_productInfo.getBundleUrl()));
+		}
+
+		return defaultBundleUrl;
+	}
+
+	private String _getDefaultReleaseDate() {
+		if (_productInfo != null) {
+			return _productInfo.getReleaseDate();
+		}
+
+		return null;
+	}
+
+	private String _getDefaultTargetplatformVersion() {
+		if (_productInfo != null) {
+			return _productInfo.getTargetPlatformVersion();
+		}
+
+		return null;
+	}
+
+	private ProductInfo _getProductInfo(String productKey) {
+		if (productKey == null) {
+			return null;
+		}
+
+		Path cacheDirPath = _cacheDir.toPath();
+
+		try {
+			Path productJsonPath = HttpUtil.downloadFile(
+				new URL(
+					_PRODUCT_INFO_URI
+				).toURI(),
+				null, null, cacheDirPath,
+				new StreamLogger() {
+
+					@Override
+					public void onCompleted() {
+					}
+
+					@Override
+					public void onProgress(long completed, long length) {
+					}
+
+					@Override
+					public void onStarted() {
+					}
+
+				});
+
+			try (JsonReader reader = new JsonReader(
+					Files.newBufferedReader(productJsonPath))) {
+
+				Gson gson = new Gson();
+
+				Map<String, ProductInfo> productInfos = gson.fromJson(
+					reader,
+					new TypeToken<Map<String, ProductInfo>>() {
+					}.getType());
+
+				if (productInfos != null) {
+					return productInfos.get(productKey);
+				}
+			}
+		}
+		catch (Exception exception) {
```

Comment:
If the user has set a product key but it doesn't exist or we couldn't download it, we should fail the build because it wont be working as user expects.